CORE-11065 `utxo_relevant_transaction_state` new columns

lankydan opened the pull request:
Store data in the `custom_representation` and `consumed` columns.

Fill `custom_representation` with temporary data.

The original `consumed` column now stores a timestamp instead of a boolean.

lankydan committed:
CORE-11065 `utxo_relevant_transaction_state` new columns

Store data in the `custom` and `consumed_timestamp` columns.

Fill `custom` with temporary data.

vlajos commented on components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoPersistenceServiceImpl.kt:
```diff
@@ -147,6 +148,7 @@ class UtxoPersistenceServiceImpl constructor(
                 UtxoComponentGroup.OUTPUTS.ordinal,
```
It is unrelated to this PR, but should `relevant` be replaced with `visible` in these parts of the code too?
Then I guess we would need a ticket for that.

lankydan replied:
@blsemo @mnesbit @MrMatthewLayton Probably.

blsemo replied:
Yes, it should be.

lankydan replied:
Shall I do this now or raise a ticket? It would be a case of renaming the table + a few methods.

blsemo replied:
Can you raise a ticket for it, please? I'd rather not have it mixed up in an unrelated PR.

vlajos commented on components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt:
```diff
@@ -202,15 +203,17 @@ class UtxoRepositoryImpl @Activate constructor(
         entityManager: EntityManager,
-        stateRefs: List<StateRef>
+        stateRefs: List<StateRef>,
+        timestamp: Instant
```
I would rename this variable to `consumedTimestamp`. Simply `timestamp` is too generic.

blsemo replied:
Interesting question - we have now a couple of operation on the `UtxoRepository` that all take a `timestamp` - and whether that's a created or consumed timestamp should be clear from the function name. So I think that is ok.

lankydan replied:
Left as a single timestamp.

vlajos commented on components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt:
```diff
@@ -331,21 +334,32 @@ class UtxoRepositoryImpl @Activate constructor(
             .setParameter("consumed", consumed)
+            .setParameter("custom", stateCustomJson.value)
             .setParameter("createdAt", timestamp)
+            .run { if (consumed) setParameter("consumedTimestamp", timestamp) else this }
```
Is it OK that `createdAt` and `consumedTimestamp` can be the same here?

lankydan replied:
I agree, I don't think there is actually a situation in our code where this happens, but I didn't want to do any refactoring as part of my change.

blsemo replied:
I don't think created and consumed timestamp being the same is a problem. Wouldn't that be the case for anything you get as part of a backchain resolution? Or do we send the actual timestamps as part of that?

lankydan replied:
The states aren't relevant/visible to you in that scenario, so you don't go into this method.

blsemo reviewed (dismissed):
Looks ok to me.

lankydan committed:
CORE-11065 PR comment - Rename column

lankydan committed:
CORE-11065 Store consumed timestamp in `consumed` column

Remove the original boolean held in the `consumed` column and store the
timestamp there instead. This is because we know a state is consumed if
this timestamp is filled in, so having the separate column was
redundant.

lankydan committed:
ALPHA

MrMatthewLayton commented on components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/utxo/tests/UtxoPersistenceServiceImplTest.kt:
```diff
@@ -128,7 +130,7 @@ class UtxoPersistenceServiceImplTest {
             serializationService = ctx.getSerializationService()
             entityManagerFactory = ctx.getEntityManagerFactory()
-            val repository = UtxoRepositoryImpl(digestService, serializationService, wireTransactionFactory)
+            repository = ctx.getSandboxSingletonService()
```
What is `ctx`?

lankydan replied:
sandbox context

blsemo reviewed (approved):
Reapproved
